README.md

```diff
@@ -1,2 +1,40 @@
-# memo
-Memo is an Anki mobile app written in Flutter
+# Memo
+
+Memo is an Anki mobile app written in Flutter. This repository holds the source code for the hybrid application - currently iOS & Android versions - of Memo.
+
+> As of now, this project is designed to only output builds for Android and iOS. Even though, given the current steps of the Flutter SDK & Dart language, there is a high possibility of compiling builds for Native desktop (Linux, Mac & Windows) & Web, but they are currently not targeted by the project.
+
+This README is meant to guide how this project is structured and should serve as a guide to help the project scale with the current and future requirements. Think of it like a flexible set of rules that guides the project's decisions. While they can (and probably will) change over the time, discussions must be raised to trigger such changes: this means that we will think/question ourselves before taking an action that breaks any rational decision taken here. It is also effective to guide PR discussions.
+
+- [Setup](#-setup): how the application environment works;
+- [Toolkit](#-toolkit): any external tools that are used to help the development process;
+- [Architecture](#-architecture): how this application works from inside;
+- [Contributing & Good Practices](#-contributing--good-practices): recommendation on how to write good code for this application;
+- [License](#-license): how this software is licensed and how you may use it.
+
+## Setup
+---
+
+### Environment
+
+### Provision Profiles and Certificates (iOS Only)
+
+### Release
+
+## Toolkit
+---
+
+## Architecture
+---
+
+The application structure and architecture is written in details in [ARCHITECTURE.md](ARCHITECTURE.md)
```

Comment:
I think we should put toolkit (or toolchain) under Architecture.
Also, I think there is no need to segment into a summary, as this README is already succinct enough.

```suggestion
## Setup

WIP

## Architecture

WIP - How this application works from inside and how it interacts with external dependencies - written in details in [ARCHITECTURE.md](ARCHITECTURE.md)
```